src/openrct2/drawing/LightFX.cpp

```diff
@@ -386,8 +382,8 @@ void LightFXPrepareLightList()
             entry->LightIntensity = static_cast<uint8_t>(
                 std::min<uint32_t>(0xFF, (entry->LightIntensity * lightIntensityOccluded) / (totalSamplePoints * 100)));
         }
-        entry->LightIntensity = static_cast<uint8_t>(
-            std::max<uint32_t>(0x00, entry->LightIntensity - static_cast<int8_t>(_current_view_zoom_front) * 5));
+        entry->LightIntensity = static_cast<uint8_t>(std::max<uint32_t>(
+            0x00, entry->LightIntensity - std::max<int8_t>(0, static_cast<int8_t>(_current_view_zoom_front)) * 5));
```

Comment:
This is _extremely_ hard to parse. Can we break this up a bit?

Reply:
I've rewritten it